client/src/components/PaymentMethods/PaymentMethods.tsx

```diff
@@ -0,0 +1,59 @@
+import { Box, Button, Typography } from '@mui/material';
+import { useStyles } from './useStyle';
+import { CardElement, useElements, useStripe } from '@stripe/react-stripe-js';
+import { PaymentMethod, StripeCardElement } from '@stripe/stripe-js';
+import { useState } from 'react';
+import PaymentCard from './PaymentCard/PaymentCard';
+import AddCardForm from './AddCardForm/AddCardForm';
+
+const PaymentMethods = (): JSX.Element => {
+  const classes = useStyles();
+  const stripe = useStripe();
+  const elements = useElements();
+  const [cardElement, setCardElement] = useState<PaymentMethod>();
+  const [openAddPayment, setOpenAddPayment] = useState(false);
+
+  const handleSubmit = async (value: { name: string }) => {
+    if (stripe && elements) {
+      await stripe
+        .createPaymentMethod({
```

Comment:
Why don't we update the snackbar with an error message in the event of a promise rejection

Reply:
add it